simulator/risc_v/riscv_register/riscv_register.h

```diff
@@ -0,0 +1,68 @@
+/* riscv_register.h - RISCV register info class 
+ * @author Alexandr Misevich
+ * Copyright 2018 MIPT-MIPS
+ */
+
+
+#ifndef RISCV_REG_H
+#define RISCV_REG_H
+
+#include <cassert>
+
+#include <array>
+#include <iostream>
+#include <utility>
+
+// MIPT-MIPS modules
+#include <infra/types.h>
+#include <infra/macro.h>
+#include <infra/string/string_view.h>
+
+class RISCVRegister {
+    enum RegNum : uint8
+    {
+#define REGISTER(X) RISCV_REG_ ## X
+#include "riscv_register.def"
+#undef REGISTER
+        , MAX_VAL_RegNum
+    };
+
+public:
+    static constexpr const size_t MAX_REG = MAX_VAL_RegNum;
+
+    explicit RISCVRegister( uint8 id) : RISCVRegister( static_cast<RegNum>( id))
+    {
+        if ( id >= 32u) {
+            std::cerr << "ERROR: Invalid RISCV register id = " << id;
+            exit( EXIT_FAILURE);
+        }
+    }
+
+    friend std::ostream& operator<<( std::ostream& out, const RISCVRegister& rhs)
+    {
+        return out << regTable[ rhs.value];
+    }
+
+    bool is_zero()       const { return value == RISCV_REG_zero; }
+    bool is_mips_hi()    const { return false; }
```

Comment:
Please add `constexpr` here so compiler would be able to do constant propagation and remove some branches while compiling RISC-V